src/server/journal/serializer.h

```diff
@@ -0,0 +1,72 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <optional>
+#include <string>
+
+#include "base/io_buf.h"
+#include "io/io.h"
+#include "server/common.h"
+#include "server/journal/types.h"
+#include "server/main_service.h"
+
+namespace dfly {
+
+// JournalWriter serializes journal entries to a sink.
```

Comment:
So I know know we have an option to write journal to file. Is this flow going to be dropped? 
If not and you will use the JournalWriter, it is writing directly to sink without any buffering

Reply:
In what sense dropped?

Its a bit more complicated. For now we can leave it as it is so the PR doesn't grow

Reply:
dropped i.e we are going to remove this flow of writing to the journal to file.
If you want to write the buffering code later maybe add TODO

Reply:
Why? The source is generic enough

For now its not part of the PR. Its a split of the original, its not finished

Reply:
I dont follow..
I dont see in the original PR buffering in the JournalWriter class
If you remember in the RDB serializer we had some buffering code which I removed cause it was redundant as we use it with StringSink and buffering is not needed
If this class is used with a sink which is a file (not a socket as in the replica flow) you will have multiple writes of 1/2/8 bytes and in this case we will need buffering in the JournalWriter or JournalWriter will write to a StringSink and there should be some wrapping code to write to the journal file.
Anyway I can wait to the next PRs and see how you use JournalWriter with File Sink

Reply:
- There is no journal file yet, I didn't cover this part at all
- In my original PR we write to a string sink in the rdb saver
- In stable state we write directly to the sink simply because I didn't yet develop a mechanism for doing it efficiently. Once we have it working and have full test coverage, we can start benchmarking it